invenio_deposit/views/rest.py

```diff
@@ -232,10 +233,15 @@ def get(self, pid, record):
     @need_record_permission('update_permission_factory')
     def post(self, pid, record):
         """Handle POST deposit files."""
-        # file name
-        key = request.form['name']
         # load the file
         uploaded_file = request.files['file']
+        # file name
+        key = secure_filename(
+            request.form.get('name') or uploaded_file.filename
+        )
+        # check if already exists a file with this name
+        if key in record.files:
+            abort(400)
```

Comment:
Instead of aborting with 400 i think we need a RESTException with a explanation of why the request failed.

```python
class FileAlreadyExists(RESTException):
    code = 400
    description = 'Filename already exists.'
```